Rename `m` to `mount`, but keep signature

stephendolan opened the pull request:
## Background and context
- Initial implementation was named `mount`, and had a signature of: `mount MyComponent.new(name: "Test")`
- Rename of `mount` -> `m` occurred, which also changed the method signature to: `m MyComponent, name: "Test"`
- #1221 opened after several issue discussions about which direction to take, deciding to revert the rename but maintain the new signature.

## Purpose
This PR renames `m` -> `mount`, retains the method signature as-is, and removes the old signature completely.

Closes #1221 

## Description
I believe I caught everything necessary. I used the regular expressions `\bm[\s(]+[A-Z]` to locate the items that needed to be changed. The only things I'd call out that need explicit review:
- I wasn't sure how to handle the old signature. Do we want deprecation of `m`, `mount` with the old signature, or do we just revert to `mount` with the new signature to get any confusion sorted during the next version bump?
- I wasn't sure whether updating the `UPGRADE.md` file was allowed, but it seemed prudent to let anyone migrating from 0.22 -> 0.23 know that they don't *really* need to rename `mount` -> `m`.

## Checklist
* [x] - An issue already exists detailing the issue/or feature request that this PR fixes
* [x] - All specs are formatted with `crystal tool format spec src`
* [x] - Inline documentation has been added and/or updated
* [x] - Lucky builds on docker with `./script/setup`
* [x] - All builds and specs pass on docker with `./script/test`

paulcsmith commented on UPGRADE_NOTES.md:
```diff
@@ -27,8 +27,7 @@ brew upgrade lucky
 
 - Update: `params.get` now strips white space. If you need the raw value, use `params.get_raw`.
-- Rename: `mount` to `m` in all pages that use components.
-- Update: all mounted components to use new signature `mount(MyComponent.new(x: 1, y: 2))` -> `m(MyComponent, x: 1, y:2)`.
```
I think we should keep this all as-is since the signature still needs to change and so that people can upgrade one version at a time if needed

stephendolan replied:
@paulcsmith Yeah, I was thinking that removing it would be okay since nothing actually _breaks_ and you just get deprecation warnings. Totally see the value in having each upgrade result in a clean, no-deprecation-warning app though. In this case if they go one by one it'll just mean a rename from `mount` -> `m` -> `mount`, but having done that myself I don't think it's too bad.

Are you good with me keeping the bolded note so that folks know it was later reverted?

paulcsmith commented on src/lucky/mount_component.cr:
````diff
@@ -6,47 +6,10 @@ module Lucky::MountComponent
-  #   text name.upcase
-  # end
-  # ```
-  @[Deprecated("Use `#m` instead. Example: m(MyComponent, arg1: 123) do/end")]
-  def mount(component : Lucky::BaseComponent) : Nil
````
Can you add a deprecated method for `m` that tells people to use `mount` instead?

stephendolan replied:
@paulcsmith For sure! I'll submit that later this week.

wontruefree commented:
should we keep the `m` alias?

stephendolan commented:
@wontruefree ,

There have been a couple issue/PR comment threads that I've read through. I was going by the comment @paulcsmith left here: https://github.com/luckyframework/lucky/issues/1049#issuecomment-661930675

I think the idea is that we don't want multiple ways to mount components since `m` is hard to search for, and may not intuitively be linked to `mount`. The option I read for handling a shorter `mount` call would be to implement an alias in your application.

stephendolan commented:
@paulcsmith Added back the `UPGRADE_NOTES` that I removed, but left the note that we reverted it. If you don't mind running through `src/lucky/mount_component.cr` one last time to make sure I added those deprecations correctly, I'd appreciate it!

Tests are all passing, though, so this should be good to merge!
